File: 5_gemini_sol1.cpp
#include <iostream>
#include <vector>
#include <string>
#include <algorithm>

using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    
    int t;
    if (!(cin >> t)) return 0;
    
    while (t--) {
        int n;
        cin >> n;
        vector<string> a(n);
        for (int i = 0; i < n; i++) {
            cin >> a[i];
        }
        
        sort(a.begin(), a.end(), [](const string& x, const string& y) {
            return x + y > y + x;
        });
        
        long long ans = 0;
        long long mod = 998244353;
        
        for (const string& s : a) {
            long long p = 1;
            long long val = 0;
            for (char c : s) {
                p = (p * 10) % mod;
                val = (val * 10 + (c - '0')) % mod;
            }
            ans = (ans * p % mod + val) % mod;
        }
        
        cout << ans << "\n";
    }
    
    return 0;
}

File: 5_gemini_sol1.py
import sys
from functools import cmp_to_key

def solve():
    input_data = sys.stdin.read().split()
    if not input_data:
        return
    
    t = int(input_data[0])
    idx = 1
    mod = 998244353
    
    def compare(x, y):
        if x + y > y + x:
            return -1
        elif x + y < y + x:
            return 1
        else:
            return 0

    out = []
    for _ in range(t):
        n = int(input_data[idx])
        idx += 1
        a = input_data[idx : idx+n]
        idx += n
        
        a.sort(key=cmp_to_key(compare))
        
        ans = 0
        for s in a:
            ans = (ans * pow(10, len(s), mod) + int(s)) % mod
        out.append(str(ans))
        
    print('\n'.join(out))

if __name__ == '__main__':
    solve()
